### PR TITLE
[ie/Piapro] Fix extractor

### LINKED ISSUE #9884: [Piapro] Unable to extract view count and content ID
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Malaysia

### Provide a description that is worded well enough to be understood

Occurs with both nightly and 2024.04.09.
Occurs with multiple songs:
* https://piapro.jp/t/5DWU
* https://piapro.jp/t/D-Gw

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
.\yt-dlp.exe -FvU "https://piapro.jp/t/D-Gw"
[debug] Command-line config: ['-FvU', 'https://piapro.jp/t/D-Gw']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version nightly@2024.05.05.232701 from yt-dlp/yt-dlp-nightly-builds [5904853ae] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.22631-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg N-104598-g8f476d5cf1-gb2aec70bd6 (setts), ffprobe N-105341-g8b6d800038-g7247a6fed8
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.02.02, curl_cffi-0.5.10, mutagen-1.47.0, requests-2.31.0, sqlite3-3.35.5, urllib3-2.2.1, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1810 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-nightly-builds/releases/latest
Latest version: nightly@2024.05.05.232701 from yt-dlp/yt-dlp-nightly-builds
yt-dlp is up to date (nightly@2024.05.05.232701 from yt-dlp/yt-dlp-nightly-builds)
[Piapro] Extracting URL: https://piapro.jp/t/D-Gw
[Piapro] D-Gw: Downloading webpage
WARNING: [Piapro] unable to extract view count; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
ERROR: [Piapro] D-Gw: Unable to extract content ID; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "yt_dlp\extractor\common.py", line 734, in extract
  File "yt_dlp\extractor\piapro.py", line 99, in _real_extract
  File "yt_dlp\extractor\common.py", line 1321, in _search_regex
```

### PR DESCRIPTION
**Pull request by FinnRG**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Piapro had a design overhaul and all extractors broke. I mostly just updated the relevant regexes.

ADD DESCRIPTION HERE

Fixes #9884
- Fixed the piapro extractor

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/piapro.py [hunk 1]
```diff
@@ -88,25 +88,19 @@ def _real_extract(self, url):
         if category_id not in ('1', '2', '21', '22', '23', '24', '25'):
             raise ExtractorError('The URL does not contain audio.', expected=True)
 
-        str_duration, str_filesize = self._search_regex(
-            r'サイズ：</span>(.+?)/\(([0-9,]+?[KMG]?B)）', webpage, 'duration and size',
-            group=(1, 2), default=(None, None))
-        str_viewcount = self._search_regex(r'閲覧数：</span>([0-9,]+)\s+', webpage, 'view count', fatal=False)
+        str_duration = self._search_regex(r'長さ：\s*(.+?)</p>', webpage, 'duration')
+        str_filesize = self._search_regex(r'サイズ：\s*([0-9.]+?[KMG]?B)', webpage, 'size')
+        str_viewcount = self._search_regex(r'閲覧数：\s*([0-9,]+)\s*</p>', webpage, 'view count', fatal=False)
```
Comment (seproDev) [inline yt_dlp/extractor/piapro.py:93]: I would slightly adjust the regex and make them all non-fatal
```suggestion
        str_duration = self._search_regex(r'長さ：\s*([\d:]+)\s*</p>', webpage, 'duration', fatal=False)
        str_filesize = self._search_regex(r'サイズ：\s*([\d.]+[KMG]?B)\s*</p>', webpage, 'size', fatal=False)
        str_viewcount = self._search_regex(r'閲覧数：\s*([\d,]+)\s*</p>', webpage, 'view count', fatal=False)
        create_date = self._search_regex(r'投稿日：\s*([\d\s/:]+)\s*</p>', webpage, 'timestamp', fatal=False)
```

File: yt_dlp/extractor/piapro.py [hunk 2]
```diff
@@ -88,25 +88,19 @@ def _real_extract(self, url):
...
 
-        uploader_id, uploader = self._search_regex(
-            r'<a\s+class="cd_user-name"\s+href="/(.*)">([^<]+)さん<', webpage, 'uploader',
-            group=(1, 2), default=(None, None))
-        content_id = self._search_regex(r'contentId\:\'(.+)\'', webpage, 'content ID')
-        create_date = self._search_regex(r'createDate\:\'(.+)\'', webpage, 'timestamp')
+        uploader_id = self._search_regex(r'<div\s+class="contents_creator">\s*<a\s+href="/(.*)"', webpage, 'uploader_id')
+        uploader = self._search_regex(r'class="contents_creator_txt">(.*)</p', webpage, 'uploader')
```
Comment (seproDev) [inline yt_dlp/extractor/piapro.py:96]: Maybe like so?
```suggestion
        uploader_id = self._search_regex(
            r'<a\s+href="/([^"]+)"', get_element_by_class('contents_creator', webpage), 'uploader id', fatal=False)
```

File: yt_dlp/extractor/piapro.py [hunk 3]
```diff
@@ -88,25 +88,19 @@ def _real_extract(self, url):
...
 
-        player_webpage = self._download_webpage(
-            f'https://piapro.jp/html5_player_popup/?id={content_id}&cdate={create_date}',
-            video_id, note='Downloading player webpage')
+        create_date = self._search_regex(r'投稿日：\s*(.+)</p>', webpage, 'timestamp')
```
Comment (seproDev) [inline yt_dlp/extractor/piapro.py:98]: Inline below
```suggestion
```

File: yt_dlp/extractor/piapro.py [hunk 4]
```diff
@@ -88,25 +88,19 @@ def _real_extract(self, url):
...
 
         return {
             'id': video_id,
-            'title': self._html_search_regex(r'<h1\s+class="cd_works-title">(.+?)</h1>', webpage, 'title', fatal=False),
-            'description': self._html_search_regex(r'(?s)<p\s+class="cd_dtl_cap">(.+?)</p>\s*<div', webpage, 'description', fatal=False),
+            'title': self._html_search_regex(r'<h1\s+class="contents_title">(.+?)</h1>', webpage, 'title', fatal=False),
+            'description': self._html_search_regex(r'(?s)<div\s+class="contents_description">(.+?)</div>\s*<div', webpage, 'description', fatal=False),
             'uploader': uploader,
```
Comment (seproDev) [inline yt_dlp/extractor/piapro.py:104]:
```suggestion
            'title': get_element_by_class('contents_title', webpage),
            'description': clean_html(get_element_by_class('contents_description', webpage)),
            'uploader': get_element_by_class('contents_creator_txt', webpage),
```

Also import `clean_html` and `get_element_by_class` from `..utils`

**Review by seproDev — CHANGES_REQUESTED** [review]
Sorry, it took so long to review

**Comment by seproDev** [comment]
I went ahead and pushed a slightly reworked version

**Review by seproDev — APPROVED** [review]

**Review by bashonly — APPROVED** [review]

**Pull request merged**